magento/pwa-studio
# Create Venia theme package

## Description

Creates a new package for the Venia theme and imports it into the existing tailwind config.

The package returns a [tailwind preset](https://tailwindcss.com/docs/presets) containing [overrides](https://tailwindcss.com/docs/theme#overriding-the-default-theme) for the default configuration (as specified in the JIRA issue).

This PR also defines the initial data for colors and functions for getting the color configuration (`getColors()`) and declaring them (`declareColors()`).

For reference, see the [POC](https://github.com/jimbo/tail/tree/main/theme) which most of the work in this PR is based on.

## Related Issue







Closes PWA-1963

## Acceptance

@jimbo 
core developers

### Verification Stakeholders

core developers

### Specification




## Verification Steps

1. Verify that yarn build succeeds
1. Verify that @magento/pwa-theme-venia package exists


#### Test scenario(s) for direct fix/feature

1. Verify new tests pass: `yarn test packages/pwa-theme-venia/lib/__tests__/* --collectCoverageFrom='packages/pwa-theme-venia/lib/**/*.js'`

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
